File: app/src/main/java/com/example/indroydlab/model/Cart_item.java
package com.example.indroydlab.model;

public class Cart_item {

        private final Product product;
        private int quantity;

        public Cart_item(Product product) {
            this.product = product;
            this.quantity = 1;
        }

        public Product getProduct() {
            return product;
        }

        public int getQuantity() {
            return quantity;
        }

        public void setQuantity(int quantity) {
            this.quantity = quantity;
        }

        public void incrementQuantity() {
            quantity++;
        }

        public double getItemPrice() {
            if (product.isDiscounted() && product.getDiscountedPrice() != null) {
                return product.getDiscountedPrice();
            }
            return product.getOriginalPrice();
        }

        public double getItemTotal() {
            double price = product.isDiscounted()
                    ? product.getDiscountedPrice()
                    : product.getOriginalPrice();
            return price * quantity;
        }

        public double getTaxAmount() {
            return getItemTotal() * product.getTaxPercent() / 100;
        }
    }

File: app/src/main/java/com/example/indroydlab/model/CartItem.kt
package com.example.indroydlab.model

class CartItem (
    val  product: ProductModel,
    var quantity: Int = 1
) {
    fun incrementQuantity() = quantity++
    fun getItemPrice(): Double {
        return if (product.isDiscounted && product.discountedPrice != null) {
            product.discountedPrice * quantity
        } else{ product.originalPrice * quantity }
    }
    fun getItemTotal(): Double {
        return if (product.isDiscounted && product.discountedPrice != null) {
            product.discountedPrice * quantity
        } else {
            product.originalPrice * quantity
        }
    }

    fun getTaxAmount(): Double{ return getItemPrice() * product.taxPercent / 100 }
}